gridcomps/History/MAPL_HistoryGridComp.F90

```diff
@@ -1269,19 +1300,86 @@ subroutine Initialize ( gc, import, dumexport, clock, rc )
+          read(tmpstring( 6: 7),'(i2.2)') month
+          read(tmpstring( 9:10),'(i2.2)') day
+          read(tmpstring(12:13),'(i2.2)') hour
+          read(tmpstring(15:16),'(i2.2)') minute
+          write(6,'(1X,"RingTime: ",i4.4, "/", i2.2, "/", i2.2, "T", i2.2, ":", i2.2, "   backwards:",L1)') &
+               year, month, day, hour, minute, list(n)%backwards
+
+          call ESMF_ClockGet(clock, currtime=debugtime, timeStep=timestep,rc=status) ; _VERIFY(STATUS)
+          call ESMF_TimeIntervalGet  ( timestep, h=hour,m=minute, rc=status ) ; _VERIFY(STATUS)
+
+          if (hour < 0 .or. minute < 0) then
+             write(6,'("Clock Timestep = -", i2.2, ":", i2.2)') hour, abs(minute)
+          else
+             write(6,'("Clock Timestep = ", i2.2, ":", i2.2)') hour, abs(minute)
+          end if
+          call ESMF_TimeGet  ( DebugTime, timeString=tmpstring, rc=status ) ; _VERIFY(STATUS)
+
+          read(tmpstring( 1: 4),'(i4.4)') year
+          read(tmpstring( 6: 7),'(i2.2)') month
+          read(tmpstring( 9:10),'(i2.2)') day
+          read(tmpstring(12:13),'(i2.2)') hour
+          read(tmpstring(15:16),'(i2.2)') minute
+          write(6,'(1X,"Current Time: ",i4.4, "/", i2.2, "/", i2.2, "T", i2.2, ":", i2.2, "   backwards:",L1)') &
+               year, month, day, hour, minute, list(n)%backwards
        endif
        if ( list(n)%backwards ) then
-          list(n)%his_alarm = ESMF_AlarmCreate( clock=clock, RingInterval=Frequency, RingTime=RingTime, rc=status )
+          list(n)%his_alarm = ESMF_AlarmCreate( clock=clock, RingInterval=Frequency, RingTime=RingTime, sticky=.false., rc=status )
        else
           list(n)%his_alarm = ESMF_AlarmCreate( clock=clock, RingInterval=Frequency, RingTime=RingTime, sticky=.false., rc=status )
        endif
        _VERIFY(STATUS)
 
+       call ESMF_ClockGetAlarmList(clock, ESMF_ALARMLIST_ALL, &
+            alarmCount=alarmCount, rc = status )
+       _VERIFY(STATUS)
+       ! WRITE(6,'("     History Clock now has ", i3, " alarms.")') alarmCount
+
+       ! I don't understand what's going on here. For some reason when I create the alarm when the clock is running
+       ! with a negative timestep, it immediately rings, even though it's an hour in the past...
+       ! if (ESMF_AlarmIsRinging(list(n)%his_alarm)) THEN
+       !    call ESMF_AlarmSet(list(n)%his_alarm, RingTime=RingTime, ringing=.false., rc=status); _VERIFY(STATUS)
+       ! endif
+       if(.false. .and.  MAPL_AM_I_ROOT() ) then
+          call ESMF_AlarmGet(list(n)%his_alarm, RingTime=DebugTime,  ringing=ringing, rc=STATUS); _VERIFY(STATUS)
+          call ESMF_TimeGet  ( DebugTime, timeString=TimeString, rc=status ) ; _VERIFY(STATUS)
+
+          read(timestring( 1: 4),'(i4.4)') year
+          read(timestring( 6: 7),'(i2.2)') month
+          read(timestring( 9:10),'(i2.2)') day
+          read(timestring(12:13),'(i2.2)') hour
+          read(timestring(15:16),'(i2.2)') minute
```

Comment:
If we cannot get these directly, we need a helper subroutine.  This logic appears 3 times in this PR.

Reply:
I see even more further down now!